src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -109,28 +111,40 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
             {
                 extensionsPath = Path.Combine(_rootScriptPath, "bin");
 
+                // Verify if the file exists and apply fallback paths
+                // The fallback order is:
+                //   1 - Script root
+                //       - If the system folder exists with metadata file at the root, use that as the base probing path
+                //   2 - System folder
                 if (!File.Exists(Path.Combine(extensionsPath, ScriptConstants.ExtensionsMetadataFileName)))
```

Comment:
Can we have all of this logic pulled out into a static method and then have a handful of tests written against it? That'd make sure we don't bust this in the future.

Reply:
Followed up offline. Additional tests, including worker integration tests will follow.